esmvalcore/_recipe.py

```diff
@@ -267,6 +280,10 @@ def _get_default_settings(variable, config_user, derive=False):
         'short_name': variable['short_name'],
         'mip': variable['mip'],
     }
+    extra_facets = get_extra_facets(variable['project'], variable['dataset'],
+                                    variable['mip'], variable['short_name'],
+                                    config_user['extra_facets_dir'])
+    fix.update(extra_facets)
```

Comment:
What @jvegasbsc  means is that instead of doing this, it would be better to just start with
```python
fix = copy.deepcopy(variable)
```
That way you have all extra facets available (because they are added to `variable` on line 1120).

Reply:
I am happy to do that. I did not do that because somebody before me had decided that it was unwise to start from the whole variable dict and specifically selected those four entries. Since I have no idea under what circumstances what other entries might land in there, I did not want to change that.

But if you two guys, @bouweandela and @jvegasbsc, think that is ok, I am happy to go along with it.

Reply:
That was because we were using only a limited amount of input to get and work the fixes, so I thought it was better to be explicit about the required keys.

This is not the first time it has to grow, so it's just better to pass everything instead of adding more and more arguments

Reply:
Ok, let's try that. It leads to two questions:

- This trips up `esmvalcore.preprocessor.check_preprocessor_settings` because we have an introspection driven whitelist of valid arguments to the preprocessor function here. Should we remove that check for these specific steps (fix_*), or maybe detect the use of `**kwargs` (simple enough: `args.varkw is None`)?
- What should we do for `add_fx_variables`? Also add the whole variable or regress to explicit `extra_facets` here?

Reply:
I have adopted the approach to pass the whole variable and have addressed the two points I raised by
- Changing the `check_preprocessor_settings` function to forego the check for exact argument matching if the preprocessor in question uses `*args` or `**kwargs`.
- Passing the entire `variable` dict also to `add_fx_variables`.

Please open a new conversation if you disagree about one of these points. Also note the changes that were necessary to the existing recipe integration tests.